core/src/debuginfo/source_manager.rs

```diff
@@ -78,10 +80,25 @@ pub enum SourceManagerError {
     /// An attempt was made to read content using invalid byte indices
     #[error("attempted to read content out of bounds")]
     InvalidBounds,
-    /// An attempt to load a source file failed due to an I/O error
-    #[cfg(feature = "std")]
-    #[error(transparent)]
-    LoadFailed(#[from] std::io::Error),
+    /// Custom error variant for implementors of the trait.
+    #[error("{error_msg}")]
+    Custom {
+        error_msg: Box<str>,
+        source: Option<Box<dyn Error + Send + Sync + 'static>>,
```

Comment:
`thiserror` returns this field for `Error::source()` when the field is named `source`, right? Can we add this as a comment, as this is not as common (at least I didn't know until reviewing this PR).